### PR TITLE
Refactor/model builder

### PR DESCRIPTION
**Pull request by Jaxing**
This will only change the basic model which we already decided to scrap. I believe that this will make it easier to implement new models as all models will use the same code and we will only have to update on one place.

### CODE REVIEW

File: model/basic/basic_model.py [hunk 1]
```diff
@@ -59,26 +56,61 @@ def __init__(self, config, session):
         self.training_epochs = config['training_epochs']
         self.users_to_select = config['users_to_select']
 
+        if 'use_l2_loss' in config:
```
Comment (hsson) [inline model/basic/basic_model.py]: With the new template system these checks can be removed as all models will have configurations for everything, even things that won't be used.

**Review by hsson — CHANGES_REQUESTED** [review]
I think this is brilliant. Though it needs to be verified that it actually works. 

Also, for this to be really useful the number of layers should also be a configuration option. Make sure to use the new config template that's been pushed to dev.

**Comment by Mxiim** [comment]
config.yaml doesn't need to be committed as it is gitignored. Would also like to see an example of how to use ModelBuilder in code, and maybe with added template to config.template.yaml if possible.

**Review by hsson — APPROVED** [review]

**Comment by hsson** [comment]
Just add new performance measure and tensorboard.

**Pull request merged**